test/Main.purs

```diff
@@ -132,3 +160,12 @@ main = run [consoleReporter] do
     it "works with Nullable" $ roundtrips (Proxy :: Proxy MyTestNullable) """
       { "a": null, "b": "a" }
     """
+    it "works with Tuple" $ roundtrips (Proxy :: Proxy MyTestTuple) """
+        { "a": [1, "foo"] }
+      """
+    it "works with nested Tuple" $ roundtrips (Proxy :: Proxy MyTestNestedTuple) """
+        { "a": [1, ["bar", 4.2]] }
```

Comment:
Test case here should be `{ "a": [1, "bar", 4.2] }`

Reply:
ah, sorry, this comment is wrong anyway

Reply:
or there's really no sane way we can try to solve nested tuples and flatten them, it seems

Reply:
Should I just close this PR? Looks like there're easy workarounds, e.g, this should serialise as [x,y,z] properly:

```
data V3 = V3 Number Number Number

derive instance gV3 :: Generic V3 _ 
instance writeForeignV3 :: WriteForeignV3 where writeImpl = genericEncode $ defaultOptions { unwrapSingleConstructors = true }
...
```

Reply:
Yeah, we could. If you need the simple pair tuple case we should put it in though.